mainnet-contracts/src/ValidatorTicket.sol

```diff
@@ -206,4 +211,102 @@ contract ValidatorTicket is
     }
 
     function _authorizeUpgrade(address newImplementation) internal virtual override restricted { }
+
+    /**
+     * @inheritdoc IValidatorTicket
+     * @dev Restricted in this context is like the `whenNotPaused` modifier from Pausable.sol
+     */
+    function purchaseValidatorTicketWithPufETH(address recipient, uint256 vtAmount)
+        external
+        virtual
+        restricted
+        returns (uint256 pufEthUsed)
+    {
+        uint256 mintPrice = PUFFER_ORACLE.getValidatorTicketPrice();
+        uint256 requiredETH = vtAmount * mintPrice / 1 ether;
+
+        uint256 pufETHToETHExchangeRate = PufferVaultV3(PUFFER_VAULT).convertToAssets(1 ether);
+        pufEthUsed = (requiredETH * 1 ether) / pufETHToETHExchangeRate;
+
+        _processPurchaseValidatorTicketWithPufETH(recipient, vtAmount, pufEthUsed);
+        return pufEthUsed;
+    }
+
+    /**
+     * @inheritdoc IValidatorTicket
+     * @dev Restricted in this context is like the `whenNotPaused` modifier from Pausable.sol
+     */
+    function purchaseValidatorTicketWithPufETHAndPermit(address recipient, uint256 vtAmount, Permit calldata permitData)
+        external
+        virtual
+        restricted
+        returns (uint256 pufEthUsed)
+    {
+        uint256 mintPrice = PUFFER_ORACLE.getValidatorTicketPrice();
```

Comment:
This is the same code as L225-L229. It can be part of the internal function to avoid code duplication